lib/hanami/configuration.rb

```diff
@@ -215,8 +219,7 @@ def load_dependent_config(require_path)
     rescue LoadError => e
       raise e unless e.path == require_path
 
-      require_relative "configuration/null_configuration"
-      NullConfiguration.new
+      raise ComponentNotAvailable, "`#{require_path}` is not available"
```

Comment:
Thanks for making this change! I feel much more comfortable now with this kind of error versus surprising NoMethodErrors on that NullConfiguration.

While we're making this change, however, could we expand on this message here to leave no room for doubt as to the reason for the error?

e.g. for this case:

```
ComponentNotAvailable: you must "hanami-router" the gem to configure `config.router` settings
```

Reply:
To make this message easier to build from within load_dependent_config, it might actually be easier now to pass it a `gem_name` instead of `require_path`, and then have it use `Hanami.bundled?` internally instead of the LoadError rescue.

Reply:
Makes sense, @timriley, I like it. Please, take a look at my last commit :pray:

Reply:
Thanks @waiting-for-dev! Just left a note for you on that new commit :)